motor/render/vulkan_renderer.cpp

```diff
@@ -629,39 +565,43 @@ class VulkanRenderer : public Renderer
     vk_pipeline_layout_ = CreatePipelineLayout(vk_device_);
     vk_graphics_pipeline_ = CreateGraphicsPipeline(
         vk_device_, vk_pipeline_layout_, vk_render_pass_);
+    vk_swaphain_framebuffers_ =
+        CreateFramebuffers(vk_device_, vk_image_views_, vk_render_pass_);
     vk_cmd_pool_ = CreateCommandPool(vk_device_,
                                      queue_indices_.graphics_queue_idx.value());
-
-    depth_buffer_ = CreateDepthBuffer(phy_dev_, vk_device_);
-
     cmd_buffers_ =
         AllocateCommandBuffers(vk_device_, vk_cmd_pool_, vk_images_.size());
+    vk_queue_ =
+        vk_device_.getQueue(queue_indices_.graphics_queue_idx.value(), 0);
 
-    vk::ClearColorValue color;
-    color.setFloat32({.0, .0, 1., .0});
-
-    vk::ImageSubresourceRange image_range;
-    image_range.setAspectMask(vk::ImageAspectFlagBits::eColor)
-        .setLevelCount(1)
-        .setLayerCount(1);
-
-    vk::CommandBufferBeginInfo cmd_buf_begin_info;
-    cmd_buf_begin_info.setFlags(
-        vk::CommandBufferUsageFlagBits::eSimultaneousUse);
-
-    for (size_t i = 0; i < cmd_buffers_.size(); ++i)
+    // Record Command Buffers
+    for (size_t i = 0; i < cmd_buffers_.size(); i++)
     {
-      auto& cmd_buffer = cmd_buffers_[i];
-      VkSuccuessOrDie(cmd_buffer.begin(cmd_buf_begin_info),
-                      "Couldn't start command buffer");
-      cmd_buffer.clearColorImage(vk_images_[i], vk::ImageLayout::eGeneral,
-                                 &color, 1, &image_range);
-
-      VkSuccuessOrDie(cmd_buffer.end(), "Couldn't end command buffer");
+      vk::CommandBufferBeginInfo begin_info;
+      VkSuccuessOrDie(cmd_buffers_[i].begin(begin_info),
+                      "Couldn't start recording command buffer");
+
+      vk::ClearColorValue clear_color;
+      clear_color.setFloat32({.6f, .6f, .6f, .0f});
+      vk::ClearValue clear_val;
+      clear_val.setColor(clear_color);
+      vk::RenderPassBeginInfo render_pass_info;
+      // Get Extent from swapchain extent;
+      render_pass_info.setRenderPass(vk_render_pass_)
```

Comment:
again can we move render_pass_info and initialization out of the loop, and only change framebuffer inside the loop?

also could we set the framebuffer before calling cmd_buffer.begin ?